dubbo-rpc/dubbo-rpc-rest/src/main/java/org/apache/dubbo/rpc/protocol/rest/filter/ServiceInvokeRestFilter.java

```diff
@@ -0,0 +1,199 @@
+        if (restMethodMetadataPair == null) {
+            throw new PathNoFoundException("rest service Path no found, current path info:" + RestRPCInvocationUtil.createPathMatcher(request));
+        }
+
+        Invoker invoker = restMethodMetadataPair.getInvoker();
+
+        RestMethodMetadata restMethodMetadata = restMethodMetadataPair.getRestMethodMetadata();
+
+        // method disallowed
+        if (!restMethodMetadata.getRequest().methodAllowed(request.getMethod())) {
+            nettyHttpResponse.sendError(405, "service require request method is : "
+                + restMethodMetadata.getRequest().getMethod()
+                + ", but current request method is: " + request.getMethod()
+            );
+            return;
+        }
+
+
+        // content-type  support judge,throw unSupportException
+        acceptSupportJudge(request, restMethodMetadata.getReflectMethod().getReturnType());
+
+        // build RpcInvocation
+        RpcInvocation rpcInvocation = RestRPCInvocationUtil.createBaseRpcInvocation(request, restMethodMetadata);
+
+        // parse method real args
+        RestRPCInvocationUtil.parseMethodArgs(rpcInvocation, request, nettyHttpRequest, nettyHttpResponse, restMethodMetadata);
+
+        // execute business  method invoke
+        Result result = invoker.invoke(rpcInvocation);
+
+        if (result.hasException()) {
+            Throwable exception = result.getException();
+            logger.error("", exception.getMessage(), "", "dubbo rest protocol provider Invoker invoke error", exception);
+
+            if (serviceDeployer.getExceptionMapper().hasExceptionMapper(exception)) {
+                writeResult(nettyHttpResponse, request, url, serviceDeployer.getExceptionMapper().exceptionToResult(result.getException()), rpcInvocation.getReturnType());
+                nettyHttpResponse.setStatus(200);
+            } else {
+                nettyHttpResponse.sendError(500,
+                    "\n dubbo rest business exception, error cause is: "
+                        + result.getException().getCause()
+                        + "\n message is: " + result.getException().getMessage()
+                        + "\n stacktrace is: " + stackTraceToString(exception));
+            }
+        } else {
+
+            // invoke the intercept chain before Result  write to  response
+            new RestResponseInterceptor() {
+                @Override
+                public void intercept(URL url, RequestFacade request, NettyHttpResponse response, Object result, RpcInvocation rpcInvocation, RestResponseInterceptorChain interceptorChain, ServiceDeployer serviceDeployer) throws Exception {
+                    interceptorChain.intercept(url, request, response, result, rpcInvocation, interceptorChain, serviceDeployer);
+                }
+            }.intercept(url, request, nettyHttpResponse, result.getValue(), rpcInvocation, new RestResponseInterceptorChain(restResponseInterceptors), serviceDeployer);
```

Comment:
1. Just RestResponseInterceptor cycle call, I think it is not necessary to increase RestResponseInterceptorChain.
2. RestResponseInterceptor should also handle exceptions, so you should not only support RestResponseInterceptor when there are no exceptions